### ISSUE TITLE
Blank rows are added in Collection for some parameters

### ISSUE THREAD

**Issue by saritabarve** [openanthem/nimbus-core#531]
# Issue Details
Blank rows are added in collection for some parameters

# How to Reproduce
An issue scenario has been created in [petclinic-training](https://github.com/openanthem/petclinic-training) for this issue. The steps to reproduce are as follows:

## Setup
1. Clone **nimbus-core** (1.3.x branch)
2. Clone [petclinic-training](https://github.com/openanthem/petclinic-training) (feature/transient-forms)
3. From the [petclinic-training](https://github.com/openanthem/petclinic-training) directory, execute: `mongo < mock-visit.js`
4. Run the **petclinic-web** application as a **Spring Boot Application**

## Reproduce in Browser
1. Navigate to http://localhost:8082/petclinic/#/h/visitlandingview/vpVisits in browser
2. Open `Visits > Visits - Client Side Link Label` in the top header
3. Select the dropdown on the right for the first `@Grid` entry and select `Pet Care Questionnaire`.
4. Expand `Section 4` to view the *Questionnaire Notes Grid*
5. Click `Add System Note` to open the *Add System Note* modal
6. Select the first entry: `Note 1 (2 owners)`
7. Click `Add Notes`
8. Select the dropdown on the right for the note just added and select `Remove`
9. Click `Add Custom`
10. Fill the required fields and click `Add`

**Observations**
1. The database is showing erroneous entries for the domain entity (see `"petcareassessment"` in mongo):
```json
{
    ...
    "petCareForm" : {
        "petCareAssessmentQuestionnaire" : {
            "questionnaire_Section4" : {
                "questionnaireNotes" : [ 
                    {
                        "_id" : NumberLong(1),
                        "content" : "some custom note"
                    }, 
                    {}, 
                    {
                        "owners" : []
                    }
                ]
            }
        }
    },
    ...
}
```

2. If repeating the steps to reproduce but in the `Add System Note` modal instead add `Note 3 (0 owners)`, **then the issue does not occur**.

**Comment by lopezton-anthem** [comment]
Thanks for your issue request.

In order to address the issue you're facing, please provide the more details on how to recreate the issue. If you're unsure of which details to provide, we've provided a handy issue template [here](https://github.com/openanthem/nimbus-core/blob/master/.github/issue_template.md).

**Comment by saritabarve** [comment]
Hi Tony,

Dileep can walk you through LTSS to demo these defects. He’s in a breakout with Mongo DB consultant till noon. Post that, he should be able to assist you.

Thanks,
Sarita

**Comment by saritabarve** [comment]
﻿Sure, Sarita. After Mongo Db meet, I will have a breakout with Tony and sync on these defects. Thank you.

Best Regards,
Dileep Roopreddy
Project Nimbus - Anthem, Inc.
dileep.roopreddy@anthem.com<mailto:dileep.roopreddy@anthem.com> | 6823651174

From: "Barve, Sarita" <sarita.barve@anthem.com>
Date: Monday, June 24, 2019 at 9:28 AM
To: openanthem/nimbus-core <reply@reply.github.com>, openanthem/nimbus-core <nimbus-core@noreply.github.com>
Cc: Author <author@noreply.github.com>, "Copley, Jaime" <jaime.copley@anthem.com>, "Roopreddy, Dileep" <dileep.roopreddy@anthem.com>, "Kotha, Chandra" <chandra.kotha@anthem.com>, "Thakur, Shambhu" <shambhu.thakur@anthem.com>, "Chigullapally, Nanda K" <nanda.chigullapally@anthem.com>, "Ushadevi, Darsana" <darsana.ushadevi@anthem.com>, "Janakarajan, Chinnasamy" <Chinnasamy.Janakarajan@anthem.com>
Subject: RE: {EXTERNAL} Re: [openanthem/nimbus-core] Blank rows are added in Collection for some parameters (#531)

Hi Tony,

Dileep can walk you through LTSS to demo these defects. He’s in a breakout with Mongo DB consultant till noon. Post that, he should be able to assist you.

Thanks,
Sarita

**Comment by saritabarve** [comment]
Thanks Dileep!

From: Roopreddy, Dileep
Sent: Monday, June 24, 2019 10:34 AM
To: Barve, Sarita <sarita.barve@anthem.com>; openanthem/nimbus-core <reply@reply.github.com>; openanthem/nimbus-core <nimbus-core@noreply.github.com>
Cc: Author <author@noreply.github.com>; Copley, Jaime <jaime.copley@anthem.com>; Kotha, Chandra <chandra.kotha@anthem.com>; Thakur, Shambhu <shambhu.thakur@anthem.com>; Chigullapally, Nanda K <nanda.chigullapally@anthem.com>; Ushadevi, Darsana <darsana.ushadevi@anthem.com>; Janakarajan, Chinnasamy <Chinnasamy.Janakarajan@anthem.com>; Ushadevi, Darsana <darsana.ushadevi@anthem.com>; Baril, David R. <david.baril@amerigroup.com>; McDonough, Antonio <tony.mcdonough@anthem.com>
Subject: Re: {EXTERNAL} Re: [openanthem/nimbus-core] Blank rows are added in Collection for some parameters (#531)

﻿Sure, Sarita. After Mongo Db meet, I will have a breakout with Tony and sync on these defects. Thank you.

Best Regards,
Dileep Roopreddy
Project Nimbus - Anthem, Inc.
dileep.roopreddy@anthem.com<mailto:dileep.roopreddy@anthem.com> | 6823651174

From: "Barve, Sarita" <sarita.barve@anthem.com<mailto:sarita.barve@anthem.com>>
Date: Monday, June 24, 2019 at 9:28 AM
To: openanthem/nimbus-core <reply@reply.github.com<mailto:reply@reply.github.com>>, openanthem/nimbus-core <nimbus-core@noreply.github.com<mailto:nimbus-core@noreply.github.com>>
Cc: Author <author@noreply.github.com<mailto:author@noreply.github.com>>, "Copley, Jaime" <jaime.copley@anthem.com<mailto:jaime.copley@anthem.com>>, "Roopreddy, Dileep" <dileep.roopreddy@anthem.com<mailto:dileep.roopreddy@anthem.com>>, "Kotha, Chandra" <chandra.kotha@anthem.com<mailto:chandra.kotha@anthem.com>>, "Thakur, Shambhu" <shambhu.thakur@anthem.com<mailto:shambhu.thakur@anthem.com>>, "Chigullapally, Nanda K" <nanda.chigullapally@anthem.com<mailto:nanda.chigullapally@anthem.com>>, "Ushadevi, Darsana" <darsana.ushadevi@anthem.com<mailto:darsana.ushadevi@anthem.com>>, "Janakarajan, Chinnasamy" <Chinnasamy.Janakarajan@anthem.com<mailto:Chinnasamy.Janakarajan@anthem.com>>
Subject: RE: {EXTERNAL} Re: [openanthem/nimbus-core] Blank rows are added in Collection for some parameters (#531)

Hi Tony,

Dileep can walk you through LTSS to demo these defects. He’s in a breakout with Mongo DB consultant till noon. Post that, he should be able to assist you.

Thanks,
Sarita

**Comment by lopezton-anthem** [comment]
I have recreated this scenario in petclinic and added the details to reproduce in the issue description.

**Comment by lopezton-anthem** [comment]
I've added a test case in [feature/NIMBUS-241](https://github.com/openanthem/nimbus-core/tree/feature/NIMBUS-241): Commit sha: 11106ea65f61a36c1bcff0ba976e0590512f1cb2

The test case is (from what I can see) a one-to-one replication of the petclinic scenario, other than the configured `SessionProvider`; yet, it is not producing the same results.

**Comment by lopezton-anthem** [comment]
This fix has deployed in version `1.3.1.M5`
